src/vs/workbench/api/worker/extHostLocalizationService.ts

```diff
@@ -0,0 +1,59 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import { Language } from 'vs/base/common/platform';
+import { URI } from 'vs/base/common/uri';
+import { IExtensionDescription } from 'vs/platform/extensions/common/extensions';
+import { AbstractExtHostLocalizationService } from 'vs/workbench/api/common/extHostLocalizationService';
+
+export class ExtHostLocalizationService extends AbstractExtHostLocalizationService {
+
+	private getBundleLocation(extension: IExtensionDescription): URI | undefined {
+		// TODO: support builtin extensions
+		// if (extension.isBuiltin && this.initData.nlsBaseUrl) {
+		// 	return URI.joinPath(this.initData.nlsBaseUrl, extension.identifier.value, 'main');
+		// }
+
+		if (extension.i18nBundleLocation) {
+			return URI.joinPath(extension.extensionLocation, extension.i18nBundleLocation);
+		}
+		return undefined;
+	}
+
+	async initializeLocalizedMessages(extension: IExtensionDescription): Promise<void> {
+		if (Language.isDefault()
+			// TODO: support builtin extensions
+			|| !extension.i18nBundleLocation
+		) {
+			return;
+		}
+
+		let contents: { [key: string]: string } | undefined;
+		const bundleLocation = this.getBundleLocation(extension);
+		if (!bundleLocation) {
+			console.error(`No bundle location found for extension ${extension.identifier.value}`);
+			return;
+		}
+		const bundleUri = URI.joinPath(bundleLocation, `bundle.i18n.${Language.value()}.json`);
+
+		try {
+			const response = await fetch(bundleUri.toString(true));
```

Comment:
Unsure about using `fetch`. We should load extension resources through the IFileService or IExtensionResourceLoaderService (on renderer side) and not use `fetch` or `readFile`. This also likely means that you need only one variant of the extHostLoc-service here

Reply:
But if we use `IFileService` or `IExtensionResourceLoaderService` then we have to get the strings in that process and then send the _potentially large payload_ over to the ExtensionHost. That sounds pretty costly, no?

Plus `IExtensionResourceLoaderService` is just a `fetch` under the hood anyway with a little bit of extra stuff wrapped around it.

Reply:
went with IFileService